apps/meteor/app/livechat/server/hooks/processRoomAbandonment.ts

```diff
@@ -43,65 +44,88 @@ const getSecondsSinceLastAgentResponse = async (room: IOmnichannelRoom, agentLas
 		officeDays = (await businessHourManager.getBusinessHour())?.workHours.reduce(parseDays, {});
 	}
 
-	if (!officeDays) {
+	if (!officeDays || Object.keys(officeDays).length === 0) {
 		return getSecondsWhenOfficeHoursIsDisabled(room, agentLastMessage);
 	}
 
 	let totalSeconds = 0;
-	const endOfConversation = moment(new Date(room.closedAt || new Date()));
-	const startOfInactivity = moment(new Date(agentLastMessage.ts));
+	const endOfConversation = moment.utc(new Date(room.closedAt || new Date()));
+	const startOfInactivity = moment.utc(new Date(agentLastMessage.ts));
 	const daysOfInactivity = endOfConversation.clone().startOf('day').diff(startOfInactivity.clone().startOf('day'), 'days');
-	const inactivityDay = moment(new Date(agentLastMessage.ts));
+	const inactivityDay = moment.utc(new Date(agentLastMessage.ts));
+
 	for (let index = 0; index <= daysOfInactivity; index++) {
 		const today = inactivityDay.clone().format('dddd');
 		const officeDay = officeDays[today];
-		const startTodaysOfficeHour = moment(`${officeDay.start.day}:${officeDay.start.time}`, 'dddd:HH:mm').add(index, 'days');
-		const endTodaysOfficeHour = moment(`${officeDay.finish.day}:${officeDay.finish.time}`, 'dddd:HH:mm').add(index, 'days');
-		if (officeDays[today].open) {
-			const firstDayOfInactivity = startOfInactivity.clone().format('D') === inactivityDay.clone().format('D');
-			const lastDayOfInactivity = endOfConversation.clone().format('D') === inactivityDay.clone().format('D');
-
-			if (!firstDayOfInactivity && !lastDayOfInactivity) {
-				totalSeconds += endTodaysOfficeHour.clone().diff(startTodaysOfficeHour, 'seconds');
-			} else {
-				const end = endOfConversation.isBefore(endTodaysOfficeHour) ? endOfConversation : endTodaysOfficeHour;
-				const start = firstDayOfInactivity ? inactivityDay : startTodaysOfficeHour;
-				totalSeconds += end.clone().diff(start, 'seconds');
-			}
+		if (!officeDay) {
+			inactivityDay.add(1, 'days');
+			continue;
+		}
+		if (!officeDay.open) {
+			inactivityDay.add(1, 'days');
+			continue;
+		}
+		if (!officeDay?.start?.time || !officeDay?.finish?.time) {
+			inactivityDay.add(1, 'days');
+			continue;
 		}
-		inactivityDay.add(1, 'days');
-	}
-	return totalSeconds;
-};
 
-callbacks.add(
-	'livechat.closeRoom',
-	async (params) => {
-		const { room } = params;
+		const officeHourStart = officeDay.start.time.split(':') as unknown as number[];
```

Comment:
What do you think of destructing the array and naming the values `officeHoursStartHour` and `officeHoursStartMinutes`? Just a tiny suggestion but It might improve readability a bit.

Reply:
def not against it 👀